s3fs/core.py

```diff
@@ -464,6 +471,19 @@ async def set_session(self, refresh=False, kwargs={}):
             return self._s3
         logger.debug("Setting up s3fs instance")
 
+        if self.session is None:
+            self.session = aiobotocore.session.AioSession(**self.kwargs)
+
+        cred_resolver = create_credential_resolver(self.session, region_name=self.session._last_client_region_used)
+        credentials = cred_resolver.load_credentials()
+
+        if credentials is None and self.key is None and self.secret is None and self.token is None and not self.anon:
+            self.anon = True
+        else:
+            self.key = credentials.access_key
```

Comment:
I added this because the values are used a couple of lines below in the init_kwargs dict. I think it should also work without because somewhere down the line it should be tried to load the credentials from configs like I did above with the `credential_resolver` but my reasoning was that credentials get loaded from configs here they can directly be supplied and don't need to be checked again later with the credentials resolver.

Reply:
Actually, is `anon=True` and there really are no creds, we get to this path too, and 
```
AttributeError: 'NoneType' object has no attribute 'access_key'
```